backend/src/books/books.service.ts

```diff
@@ -1,13 +1,60 @@
 import { Injectable, HttpService } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import * as Papa from 'papaparse';
+import { Readable } from 'stream';
+import { validate, ValidationError } from 'class-validator';
 
-import { BookQueryResponse } from './books';
-import { Book } from '../../../interfaces/books';
+import { BookQueryResponse, BookUploadResult } from './books';
+import { Book as BookType } from '../../../interfaces/books';
+import { Book } from './book.entity';
+import { BookDto } from './book.dto';
+
+/** https://stackoverflow.com/a/54136803 */
+const bufferToStream = (buffer: Buffer): Readable => {
```

Comment:
Out of curiosity, would this work if its inside the class and written as a class function? 🤔